src/Tizen.TV.UIControls.Forms/Renderer/GridViewRenderer.cs

```diff
@@ -0,0 +1,205 @@
+            }
+            else if (e.PropertyName == GridView.VerticalScrollBarVisibleProperty.PropertyName)
+            {
+                Control.VerticalScrollBarVisiblePolicy =
+                    Element.VerticalScrollBarVisible ? ElmSharp.ScrollBarVisiblePolicy.Visible : ScrollBarVisiblePolicy.Invisible;
+            }
+            else if (e.PropertyName == GridView.OrientationProperty.PropertyName)
+            {
+                Control.IsHorizontal = Element.Orientation == GridViewOrientation.Horizontal ? true : false;
+            }
+            else if (e.PropertyName == GridView.ItemsSourceProperty.PropertyName)
+            {
+                UpdateItemsSource();
+            }
+            else if (e.PropertyName == GridView.SelectedItemProperty.PropertyName)
+            {
+                UpdateSelectedItem();
+            }
+            else if (e.PropertyName == GridView.ThemeStyleProperty.PropertyName)
+            {
+                Control.Style = Element.ThemeStyle;
+            }
+            base.OnElementPropertyChanged(sender, e);
+        }
+
+        void OnItemSelected(object sender, GenGridItemEventArgs e)
+        {
+            GengridItemContext context = e.Item.Data as GengridItemContext;
+            Element.SelectedItem = context.Data;
+        }
+
+        void OnItemUnfocused(object sender, GenGridItemEventArgs e)
+        {
+            GengridItemContext context = e.Item.Data as GengridItemContext;
+            (Element as IGridViewController)?.SendItemFocused(new GridViewFocusedEventArgs(context.Data, context.RealizedView, false));
+        }
+
+        void OnItemFocused(object sender, GenGridItemEventArgs e)
+        {
+            GengridItemContext context = e.Item.Data as GengridItemContext;
+            (Element as IGridViewController)?.SendItemFocused(new GridViewFocusedEventArgs(context.Data, context.RealizedView, true));
+        }
+
+        void UpdateItemsSource()
+        {
+            _genGrid.Clear();
+            itemContexts.Clear();
+            _genGridItemDic.Clear();
+
+            if (Element.ItemsSource == null)
+            {
+                return;
+            }
+
+            foreach (var item in Element.ItemsSource)
+            {
+                View realview = ContentCreator.CreateContent(Element.ItemTemplate, item, Element.ItemHeight, Element.ItemWidth);
+                realview.BindingContext = item;
+                var context = new GengridItemContext
+                {
+                    Data = item,
+                    RealizedView = realview,
+                };
+                itemContexts.Add(context);
+                var gridItem = _genGrid.Append(gridItemClass, context);
+                _genGridItemDic.Add(item, gridItem);
+            }
+
+            if (Element.ItemsSource is INotifyCollectionChanged collection)
+            {
+                collection.CollectionChanged += OnCollectionChanged;
+            }

```

Comment:
Need to unset event handler when ItemsSources was changed, 

```suggestion
            if (_collection != null) {
                  _collection.CollectionChanged -= OnCollectionChanged;
                  _collection = null;
             }
            if (Element.ItemsSource is INotifyCollectionChanged collection)
            {
                _collection = collection;
                collection.CollectionChanged += OnCollectionChanged;
            }

```